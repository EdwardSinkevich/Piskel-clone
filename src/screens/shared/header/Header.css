.nav-btn {
  margin-right: 6px;
  padding: 5px 10px;
  border-style: none;
  border-radius: 2px;
  cursor: pointer;
  font-size: 13px;
  font-weight: bold;
  text-align: center;
  color: black;
  background-color: gold;
  transition: 0.3s background-color;
}

.navigation {
  display: flex;
  justify-content: space-between;
  height: 36px;
  line-height: 36px;
  background: #2d2d2d;
  border-bottom: 1px solid #1a1a1a;
}

.piskel-name {
  margin-left: 10px;
  font-size: 16pt;
  color: #888;
  text-shadow: 0 -1px 0 #000;
}

.google-sign-in-btn {
  padding: 5px 25px;
  background-color: white;
  background-image: url('./images/google.png');
  background-position: 3% 50%;
  background-size: 25%;
  background-repeat: no-repeat;
}

.user-data {
  display: inline;
}

.user-avatar {
  width: 20px;
  height: 20px;
  position: relative;
  top: 5px;
}
