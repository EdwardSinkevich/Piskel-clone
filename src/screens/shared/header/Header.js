import './Header.css';

export default class Header {
  static render() {
    const header = document.getElementById('header');
    header.innerHTML = `
      <div class="navigation">
        <span class="piskel-name">Piskel Clone</span>
        <nav id="navigation-links" class="navigation-links">
          <a id="customBtn" class="nav-btn google-sign-in-btn" href="#">Sign in</a>
        </nav>
      </div>`;
  }
}
