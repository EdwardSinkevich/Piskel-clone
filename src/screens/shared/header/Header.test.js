import Header from './Header';

describe('Header.render', () => {
  it('Should create Header correctly', () => {
    const header = document.createElement('header');
    header.id = 'header';
    document.body.appendChild(header);
    Header.render();
    expect(document.getElementById('header').innerHTML).toMatchSnapshot();
  });
});
