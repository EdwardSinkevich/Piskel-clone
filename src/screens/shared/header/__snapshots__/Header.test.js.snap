// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.render Should create Header correctly 1`] = `
"
      <div class=\\"navigation\\">
        <span class=\\"piskel-name\\">Piskel Clone</span>
        <nav id=\\"navigation-links\\" class=\\"navigation-links\\">
          <a id=\\"customBtn\\" class=\\"nav-btn google-sign-in-btn\\" href=\\"#\\">Sign in</a>
        </nav>
      </div>"
`;
